[Li Shihao] iP

l-shihao opened the pull request:
Append part of my initial file structure set-up of the project in README.md :
# Set up
Current folder set up as below, shows java files and class files location:
```
├── out
│   └── production
│       └── ip
│           ├── duke.Duke.class
│           └── duke
│               ├── ...
│      
├── src
│   └── main
│       └── java
│           └── duke
│               ├── Duke.java
│               ├── command
│               │   ├── Command.java
│               │   ├── CommandCaller.java
│               │   ├── CommandFactory.java
│               │   ├── DLCreationCmd.java
│               │   ├── EventCreationCmd.java
│               │   ├── TaskMarkDoneCmd.java
│               │   ├── TodoCreationCmd.java
│               │   └── UndoableCommand.java
│               ├── parse
│               │   └── StringParser.java
│               ├── storage
│               │   ├── TempList.java
│               │   └── TempTaskList.java
│               ├── task
│               │   ├── Deadline.java
│               │   ├── Event.java
│               │   ├── Task.java
│               │   └── Todo.java
│               └── ui
│                   └── Message.java
```

`tree` command can be used to reveal above folder structure. 

To compile the java files, run below command (change once new packages created): 
```
% javac -d out/production/ip src/main/java/duke/Duke.java src/main/java/duke/ui/* src/main/java/duke/storage/* 
```

To run the program in terminal. `cd` to project root dir, then run below: 
```
% java -cp out/production/ip/ duke.Duke
```


# Aspiration
My own new implementation of the Duke project. Want to try out new things 
- [x] Command Pattern 
- [x] EventListener Pattern
- [ ] Lambda Expression
- [ ] Optional

> All race conditions, deadlock conditions, and concurrent update problems are due to mutable variables. 
> ― Robert C. Martin, Clean Architecture

Commit:
Add Gradle support

l-shihao committed:
Update README.md

why seems cannot edit in IntelliJ IDEA

l-shihao committed:
First edit, new start, Level-1

l-shihao committed:
Merge branch 'master' of https://github.com/l-shihao/ip
merge README.md

l-shihao committed:
Level-2, echo/list tasks, Implicit Invocation

l-shihao committed:
Level-3: Mark as Done, A-Classes: Task Class

l-shihao committed:
Minimal commit: CommandFactory, Command Pattern, Level-4

l-shihao committed:
Level-5 Handle Errors, Exception, Branch

l-shihao committed:
Merge tag 'level-5'

Level-5 Handle Errors

sohwkjames commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+ * @since       2021 Aug
+ */
+public class Duke {
+    private TempTaskList<Task> list = new TempTaskList<>();
```
I think this is a bit tight coupling

sohwkjames commented on src/main/java/duke/command/CommandCaller.java:
```diff
@@ -0,0 +1,45 @@
+
+    public CommandCaller() {}
+    public CommandCaller(CommandFactory commandFactory) {
+        this.commandFactory = commandFactory;
```
Good usage of design patterns here

sohwkjames reviewed (commented):
Good usage of design patterns! An area of improvement is there is some tight coupling with the task list. Looks good to me overall

badfr0g commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,30 @@
+        super(title);
+        this.at = at;
+    }
+
+    public String getStartTime() {
+        return at;
+    }
+    public LocalDateTime getEndTime() {
+        return till;
+    }
+
+
+
+
+    @Override
+    public String toString() {
+        return taskNo + ". " + "[E]" + super.toString()
+                + " (at: " + at + ")";
+    }
```
i think it will be good to keep the space between methods consistent

yaololo commented on src/main/java/duke/command/CommandFactory.java:
```diff
@@ -0,0 +1,43 @@
+import java.io.File;
+import java.util.HashMap;
+
+public class CommandFactory {
```
I like the `CommandFactory` concept to keep all available commands together. 👍

l-shihao committed:
coding standard, quality done along the way

l-shihao committed:
Merge branch 'branch-a-code-quality'

l-shihao committed:
Abstract Class

l-shihao committed:
Merge branch 'branch-a-abstractclass'

l-shihao committed:
packages creation along the way

l-shihao committed:
Merge branch 'branch-a-packages'

l-shihao committed:
Level-6, delete multiple tasks at once

l-shihao committed:
Merge pull request #1 from l-shihao/branch-Level-6

Level-6, Delete cmd for deletion of  multiple tasks at once

l-shihao committed:
A-TextUiTesting, find files under ./test-ui-testing

l-shihao committed:
Branch for Level-7

l-shihao committed:
A-JavaDoc

l-shihao committed:
Merge branch 'branch-level-7'

l-shihao committed:
Branch-A-JavaDoc

l-shihao committed:
try resolve conflict with branch-a-javadoc

l-shihao committed:
try resolve conflict with branch-a-javadoc1

l-shihao committed:
bug fix: remove start up tasks add msg, extra spaces introduced to the task title

l-shihao committed:
Level-8, date & time

l-shihao committed:
Level-9 Find

l-shihao committed:
Merge pull request #2 from l-shihao/branch-Level-8

Level-8, date & time 
Merge to `master` branch

l-shihao committed:
Merge branch 'branch-Level-9'

l-shihao committed:
A-JUnit. Add JUnit tests

l-shihao committed:
Merge branch 'branch-A-JUnit'

l-shihao committed:
Bug Fix: done status not write to disk

l-shihao committed:
Set theme jekyll-theme-cayman

l-shihao committed:
Update User Guide 

which is under ./docs/README.md

l-shihao committed:
wait for merging with branch-A-UG

l-shihao committed:
solve merge conflict 1

l-shihao committed:
Fix the section links

l-shihao committed:
Fix section links

l-shihao committed:
Update README.md, fix text-ui-test

l-shihao committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into branch-a-gradle

l-shihao committed:
to merge branch-a-checkstyle

l-shihao committed:
Merge remote-tracking branch 'origin/branch-a-checkstyle'

l-shihao committed:
add .github/workflows/gradle.yml GH Action

l-shihao committed:
GUI Support; README update

l-shihao committed:
fix README img link

l-shihao committed:
fix README link; remove yml IO redirect

l-shihao committed:
fix checkstyle

l-shihao committed:
Rename DLCreationCmd.java to DlCreationCmd.java

My local file is the correct name

l-shihao committed:
Fix css UI bug, replace emoji text to image; git ignore remove config/ for CI

l-shihao committed:
support to add tags during adding task

l-shihao committed:
Extension: C-Tagging

l-shihao committed:
Add docs/UI.png

l-shihao committed:
Edit docs/README.md for testing layout

l-shihao committed:
Update docs/UserGuide.md

l-shihao committed:
Rename UI.png to Ui.png

l-shihao committed:
Version 0.5.1 for A-Release
Main update on Documentation:

+ Add Class Diagram and Sequence Diagram to DG.md
+ Fix header links in UG.md

l-shihao committed:
Fix build.gradle plugin JavaFX dependencies conflicts
